### PR TITLE
feat(openpi): support pi0 and pi0.5 convert and deployment

### PR DESCRIPTION
**Pull request by Code-Hit-ai**
This PR updates the OpenPI checkpoint conversion workflow and adds support for Pi0 and Pi0.5 real-world deployment in two checkpoint formats:

1. Legacy OpenPI PyTorch deployment using `full_weights.pt`
2. OpenPI_RLinf deployment using `model.safetensors`

It also adds the corresponding Pi0 and Pi0.5 Dual-Franka SFT/evaluation configurations, introduces configurable deployment precision, and synchronizes the English and Chinese documentation.

### Checkpoint conversion

- Add `pt_to_realworld_pt.py` for directly converting RLinf Pi0 and Pi0.5 SFT checkpoints to the legacy real-world `full_weights.pt` format.
- Keep the PT-to-PT conversion entirely in FP32 without creating an intermediate BF16 checkpoint.
- Automatically detect whether the source checkpoint is Pi0 or Pi0.5.
- Automatically select the corresponding normalization mapping:
  - Standard RMSNorm mapping for Pi0
  - Adaptive RMSNorm mapping for Pi0.5
- Validate source precision, model variant, parameter names, tensor shapes, and output precision.
- Use a matching reference model to provide the expected deployment schema and the action-expert language-model head.
- Generate a JSON conversion report alongside the converted checkpoint.
- Replace the previous `sft2deploy.py` entry.
- Register the converter as `pt_to_realworld_pt` in the unified `convert.py` dispatcher.
- Export the new converter from the OpenPI converter package.
- Update the checkpoint converter README with supported formats, validation behavior, and usage examples.

### Pi0 and Pi0.5 SFT configurations

- Rename the Dual-Franka Pi0.5 SFT configuration to:

  `realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml`

- Add the Dual-Franka Pi0 SFT configuration:

  `realworld_sft_openpi_rlinf_pi0_dual_franka_tcp_rot6d.yaml`

- Add the Pi0 Dual-Franka model configuration:

  `examples/embodiment/config/model/pi0_rlinf.yaml`

- Register the `pi0_dualfranka_tcp_rot6d` OpenPI data configuration.
- Keep the existing upstream SFT model templates unchanged.

### Real-world deployment configurations

Add legacy OpenPI PyTorch and OpenPI_RLinf deployment configurations for both Pi0 and Pi0.5.

#### Pi0

- `realworld_eval_dual_franka_openpi_pi0.yaml`
  - Deploys the legacy FP32 `full_weights.pt` format.
- `realworld_eval_dual_franka_openpi_pi0_rlinf.yaml`
  - Deploys the OpenPI_RLinf `model.safetensors` format.

#### Pi0.5

- `realworld_eval_dual_franka_openpi_pi05.yaml`
  - Deploys the legacy FP32 `full_weights.pt` format.
- `realworld_eval_dual_franka_openpi_pi05_rlinf.yaml`
  - Deploys the OpenPI_RLinf `model.safetensors` format.

### Deployment precision

- Add deployment-side precision handling through `rollout.model.precision`.
- Support the following values:
  - `fp32`: use pure FP32 precision
  - `null`: use OpenPI's default mixed precision
  - `bf16`: use pure BF16 precision
- Set the new Dual-Franka Pi0 and Pi0.5 deployment configurations to FP32 by default.
- Update the corresponding Pi0 and Pi0.5 OpenPI_RLinf model configurations to use FP32 by default.

### Documentation

- Rename the Dual-Franka OpenPI PyTorch guide to the OpenPI_RLinf guide.
- Extend the guide to cover both Pi0 and Pi0.5.
- Document both legacy `full_weights.pt` and OpenPI_RLinf `model.safetensors` deployment workflows.
- Add Pi0 and Pi0.5 SFT, conversion, and deployment commands.
- Document the FP32 PT-to-PT conversion behavior and automatic model-vari

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst [hunk 1]
```diff
@@ -79,6 +87,31 @@ Observation and Action
...
+   * - Item
+     - Pi0
+     - Pi0.5
+   * - Continuous state in model core
+     - Yes
+     - No
+   * - SFT entry and checkpoint layout
+     - Shared
+     - Shared
+   * - Legacy PT converter
```
Comment (qurakchin) [inline docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst:108]: Why keep the old version openpi pytorch?

File: docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst [hunk 2]
```diff
@@ -55,11 +60,14 @@ Tasks
...
    * - SFT
-     - ``realworld_sft_openpi_pytorch_dual_franka_tcp_rot6d``
-     - Fine-tune π₀.₅ on tcp_rot6d actions.
-   * - Deployment
-     - ``realworld_eval_dual_franka``
-     - Run eval-only deployment on the robot nodes.
+     - ``realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d``
+     - Fine-tune Pi0 or Pi0.5 on tcp_rot6d actions.
+   * - Deployment (OpenPI PyTorch)
+     - ``realworld_eval_dual_franka_openpi_pi05``
```
Comment (qurakchin) [inline docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst]: Why keep the old version openpi pytorch?

File: docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst (hunk 1 above)
Comment (Code-Hit-ai) [reply docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst:108]: Because the difference between the new and old versions on the deployment side is actually the format of pt and safetensors, only one convert is needed to support it, without the need to modify too much deployment side code

File: docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst (hunk 2 above)
Comment (Code-Hit-ai) [reply docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst]: Because the difference between the new and old versions on the deployment side is actually the format of pt and safetensors, only one convert is needed to support it, without the need to modify too much deployment side code

Comment (qurakchin) [reply docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst]: It seems that the new deployment code and config is ready in "realworld_eval_dual_franka_openpi_pi05_rlinf.yaml". If the new version is ready and correct, the old version can be removed.

File: examples/sft/config/realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ hydra:
   searchpath:
```
Comment (qurakchin) [inline examples/sft/config/realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml:11]: what's the difference in this file

Comment (qurakchin) [reply examples/sft/config/realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml:11]: is the model_type: "openpi_rlinf" right in the pi0_5_pytorch_dual_franka

File: docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst (hunk 2 above)
Comment (Code-Hit-ai) [reply docs/source-en/rst_source/examples/embodied/dual_franka_openpi_rlinf.rst]: We tested it on real machines and found that there may be missing points in the weight format of the new version

File: examples/sft/config/realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml (hunk 3 above)
Comment (Code-Hit-ai) [reply examples/sft/config/realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml:11]: Openpi_rlinf is our new implementation, and I have standardized the naming here. The code was synchronized and changed before

File: examples/embodiment/config/model/pi0_5_rlinf.yaml [hunk 4]
```diff
@@ -4,8 +4,8 @@ model_type: "openpi_rlinf"
 # eval experiment config (behavior_openpi_pi05_rlinf_eval.yaml); the values here
 # are placeholders so this template stays path-free and reusable.
 
-# Weight dtype for eval: bf16. (SFT keeps an fp32 master; see the SFT template.)
-precision: bf16
+# Weight dtype for eval. Dual-Franka deployment defaults to full precision.
+precision: fp32
```
Comment (guozhen1997) [inline examples/embodiment/config/model/pi0_5_rlinf.yaml]: The comment here is confusing, please remove it. The default precision should be `bf16`, please do not change this value.

File: examples/embodiment/config/model/pi0_rlinf.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+model_type: "openpi_rlinf"
+
+# OpenPI_RLinf Pi0 model shape for Dual-Franka tcp_rot6d evaluation. The
+# checkpoint path is supplied by the real-world evaluation config.
+model_path: "/path/to/model/openpi_rlinf"
+precision: fp32
```
Comment (guozhen1997) [inline examples/embodiment/config/model/pi0_rlinf.yaml]: Please use `bf16` as the default precision.

File: rlinf/models/embodiment/openpi/__init__.py [hunk 6]
```diff
@@ -87,7 +87,14 @@ def get_model(cfg: DictConfig, torch_dtype=None):
             all_state_dict.update(state_dict)
         model.load_state_dict(all_state_dict, strict=False)
 
-    model.paligemma_with_expert.to_bfloat16_for_selected_params("bfloat16")
+    if torch_dtype == torch.float32:
```
Comment (guozhen1997) [inline rlinf/models/embodiment/openpi/__init__.py]:
```
if torch_dtype is not None:
    model = model.to(dtype=torch_dtype)
else:
    # Preserve the existing OpenPI mixed-precision behavior for
    # precision=null configurations.
    model.paligemma_with_expert.to_bfloat16_for_selected_params("bfloat16")
```

File: examples/sft/config/realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml
Comment (guozhen1997) [inline examples/sft/config/realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml]: what is the difference between `pi0_5`, `pi0_5_rlinf` and `pi0_5_pytorch_dual_franka`.

File: rlinf/utils/ckpt_convertor/openpi/pt_to_realworld_pt.py
Comment (guozhen1997) [inline rlinf/utils/ckpt_convertor/openpi/pt_to_realworld_pt.py:1]: This file converts the new openpi_rlinf full_weights.pt to openpi_pytorch format  it has nothing to do with the realworld. I suggest merge this file into `openpi_rlinf_to_openpi_pytorch.py`.

File: examples/embodiment/config/model/pi0_5_rlinf.yaml (hunk 4 above)
Comment (Code-Hit-ai) [reply examples/embodiment/config/model/pi0_5_rlinf.yaml]: Thanks for pointing this out. Done

File: examples/embodiment/config/model/pi0_rlinf.yaml (hunk 5 above)
Comment (Code-Hit-ai) [reply examples/embodiment/config/model/pi0_rlinf.yaml]: Thanks for pointing this out. Done

File: rlinf/models/embodiment/openpi/__init__.py (hunk 6 above)
Comment (Code-Hit-ai) [reply rlinf/models/embodiment/openpi/__init__.py]: Thanks for the suggestion. Done — the precision handling has been simplified as suggested.

File: rlinf/utils/ckpt_convertor/openpi/pt_to_realworld_pt.py
Comment (Code-Hit-ai) [reply rlinf/utils/ckpt_convertor/openpi/pt_to_realworld_pt.py:1]: Thanks for pointing this out. Done — the FP32 PT conversion has been merged into `openpi_rlinf_to_openpi_pytorch.py`, and the separate file and mode have been removed.

File: examples/sft/config/realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml
Comment (Code-Hit-ai) [reply examples/sft/config/realworld_sft_openpi_rlinf_pi05_dual_franka_tcp_rot6d.yaml]: Thanks for the suggestion. `pi0_5` uses the legacy OpenPI implementation, while `pi0_5_rlinf` uses OpenPI_RLinf. Done — the Dual-Franka config now reuses `pi0_5_rlinf` and overrides its task-specific fields.

File: rlinf/models/embodiment/openpi/__init__.py [hunk 7]
```diff
@@ -87,7 +87,12 @@ def get_model(cfg: DictConfig, torch_dtype=None):
             all_state_dict.update(state_dict)
         model.load_state_dict(all_state_dict, strict=False)
 
-    model.paligemma_with_expert.to_bfloat16_for_selected_params("bfloat16")
+    if torch_dtype is not None:
```
Comment (F9rozen) [inline rlinf/models/embodiment/openpi/__init__.py:90]: openpi_rlinf/__init__.py has no such branch: null leaves the Pi0Config create dtype (bfloat16). Please split the precision table by backend, or drop null from the OpenPI_RLinf instructions.

Comment (Code-Hit-ai) [reply rlinf/models/embodiment/openpi/__init__.py:90]: Thanks for pointing this out. Done — I split the precision documentation by backend in both the English and Chinese guides. The table now clarifies that `null` selects mixed precision for legacy OpenPI, while OpenPI_RLinf keeps the BF16 dtype created by `Pi0Config`.
